openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -211,6 +212,33 @@ class LibraryTeamView(APIView):
     Note also the 'allow_public_' settings which can be edited by PATCHing the
     library itself (LibraryDetailsView.patch).
     """
+    @convert_exceptions
+    def post(self, request, lib_key_str):
+        """
+        Add a user to this content library via email, with permissions specified in the
+        request body.
+        """
+        key = LibraryLocatorV2.from_string(lib_key_str)
+        api.require_permission_for_library_key(key, request.user, permissions.CAN_EDIT_THIS_CONTENT_LIBRARY_TEAM)
+        serializer = ContentLibraryAddPermissionByEmailSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        try:
+            user = User.objects.get(email=serializer.validated_data.get('email'))
+        except User.DoesNotExist:
+            raise ValidationError({'email': _('We could not find a user with that email address.')})
+        grant = api.get_library_user_permissions(key, user)
+        if grant:
+            return Response(
+                {'email': [_('This user already has access to this library.')]},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        try:
+            api.set_library_user_permissions(key, user, access_level=serializer.validated_data["access_level"])
+        except ValueError as err:
+            raise ValidationError(detail=str(err))
```

Comment:
Catches of Key/Value/Attribute/Type errors always give me pause -- if they are unexpected, I'd rather let the 500 just happen, which increases their visibility to operators.

Can you either make the catch more specific, include a comment of what you expect this to catch, or just let the exception happen? Same goes for the instances below.